src/braket/experimental/autoqasm/program/program.py

```diff
@@ -107,6 +108,15 @@ def qubits(self) -> List[int]:
         # Can be memoized or otherwise made more performant
         return sorted(list(self._qubits_seen))
 
+    @property
+    def is_in_main(self) -> bool:
+        """Whether the main block is currently being processed."""
+        return self._in_main
+
+    def in_main(self, in_main: bool) -> None:
+        """Set context for processing the `main` block."""
+        self._in_main = in_main
+
```

Comment:
Do you get a benefit from making this a property rather than just using/accessing it as a simple variable `self.is_main`?

Reply:
As a property, no, except for documentation, but the setter syntax becomes a function call. I was on the fence about this myself, I preferred `context.in_main(True)` but only stylistically. I'm happy to update it as a simple variable

Reply:
Documentation is a good point. I'm indifferent, just curious.

Reply:
Turns out it's not necessary anyway, because your rec about `subroutines_processing` worked great!